projects/igniteui-angular/src/lib/grids/filtering/grid-filtering-row.component.ts

```diff
@@ -527,44 +534,55 @@ export class IgxGridFilteringRowComponent implements AfterViewInit {
         this._cancelChipClick = chip.selected && activeElement && this.inputGroup.nativeElement.contains(activeElement);
     }
 
-    public onChipClick(args, chip: IgxChipComponent) {
+    public onChipClick(args, item: ExpressionUI) {
         if (this._cancelChipClick) {
             return;
         }
 
         this._cancelChipClick = false;
-        chip.selected = !chip.selected;
-    }
+        item.isSelected = !item.isSelected;
 
-    /**
-     *  Event handler for chip selected event.
-     */
-    public onChipSelected(eventArgs: IChipSelectEventArgs, expression: IFilteringExpression) {
-        if (eventArgs.selected) {
-            if (this.chipsArea.chipsList) {
-                this.chipsArea.chipsList.forEach((chip) => {
-                    if (chip !== eventArgs.owner) {
-                        chip.selected = false;
+        if (item.isSelected) {
+            if (this.expressionsList) {
+                this.expressionsList.forEach((expr) => {
+                    if (expr !== item) {
+                        expr.isSelected = false;
                     }
                 });
             }
-            this.expression = expression;
+            this.expression = item.expression;
 
             if (this.input) {
                 this.input.nativeElement.focus();
             }
-        } else if (this.expression === expression) {
+        } else if (this.expression === item.expression) {
             this.resetExpression();
         }
     }
 
+    /**
+     *  Event handler for chip selected event.
+     */
+    public onChipSelected(eventArgs: IChipSelectEventArgs) {
```

Comment:
Do we need this method? We should extract a `toggleSelection` method which should be called in `onChipClick` and `onChipKeyDown` in order to avoid code duplication.

Reply:
Selection handler is now removed